### ISSUE TITLE
On designing a V module

### ISSUE THREAD

**Issue by zuoanqh** [worldmodels/worldmodels.github.io#4]
Greetings and thanks for the wonderful paper.

I was reading how you want to use an unsupervised segmentation layer, and this idea came to my mind. I see you cited **Curiosity-driven Exploration by Self-supervised Prediction**, which used an inverse dynamic task (from the learn to poke by poking paper) instead of pixel reconstruction task here. My first reaction is, this begs the question: which is better for the V module?

I know there are a couple of studies such as **Integrating State Representation Learning into Deep Reinforcement Learning**, **Loss is its own Reward: Self-Supervision for Reinforcement Learning** which compared different representation learning tasks. In these two, VAE has performed worse than inverse dynamic (sometimes VAE (but not BiGAN, thinking emoji) task even degrades the performance). 

However, I have not seen a study comparing them for the purpose of constructing a world model. It would be nice if there is a paper investigating this problem. Or is there one already, I just haven't found it?

**Comment by hardmaru** [comment]
Hi @zuoanqh 

Thanks for the comments. If your state space is not high dimensional (like images), can you consider just using the M model directly to estimate the state vector, using the MDN? Just make sure to scale and normalise the state data to a range of sane values (like normalising to around zero mean unit variance for each dimension).

**Comment by zuoanqh** [comment]
Ah, I'm considering pixel inputs, just as the papers I referenced are. But what you said is a good idea, I'll consider it if I start working with other inputs one day.

**Comment by hardmaru** [comment]
Yeah, I feel with environments that have state information, it might be worthwhile to try just C-M first to make sure it works, so if the C-V-M thing doesn't work later it will be easier to debug and isolate the problems, and allows you to more confidently try different candidates for V (like those that you listed above).
